### PR TITLE
Added const to moveit_core/collision_detection per issue 879

### PR DESCRIPTION
**Pull request by bgill92**
### Description

Const'd function parameters and locally scoped variables in the collision_detection portion of of moveit_core. This PR pertains to [Const as much as possible #879](https://github.com/ros-planning/moveit2/issues/879) 

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_core/collision_detection/src/collision_matrix.cpp [hunk 1]
```diff
@@ -135,14 +135,14 @@ bool AllowedCollisionMatrix::hasEntry(const std::string& name) const
 
 bool AllowedCollisionMatrix::hasEntry(const std::string& name1, const std::string& name2) const
 {
-  auto it1 = entries_.find(name1);
+  const auto it1 = entries_.find(name1);
   if (it1 == entries_.end())
     return false;
   auto it2 = it1->second.find(name2);
```
Comment (vatanaksoytezer) [inline moveit_core/collision_detection/src/collision_matrix.cpp:141]:
```suggestion
  const auto it2 = it1->second.find(name2);
```

File: moveit_core/collision_detection/src/collision_matrix.cpp [hunk 2]
```diff
@@ -105,7 +105,7 @@ AllowedCollisionMatrix::AllowedCollisionMatrix(const moveit_msgs::msg::AllowedCo
 
 bool AllowedCollisionMatrix::getEntry(const std::string& name1, const std::string& name2, DecideContactFn& fn) const
 {
-  auto it1 = allowed_contacts_.find(name1);
+  const auto it1 = allowed_contacts_.find(name1);
   if (it1 == allowed_contacts_.end())
     return false;
   auto it2 = it1->second.find(name2);
```
Comment (vatanaksoytezer) [inline moveit_core/collision_detection/src/collision_matrix.cpp:111]:
```suggestion
  const auto it2 = it1->second.find(name2);
```

**Review by vatanaksoytezer — APPROVED** [review]
Just a few more places to put consts, otherwise lgtm! Thank you @bgill92!

File: moveit_core/collision_detection/src/collision_env.cpp [hunk 3]
```diff
@@ -42,7 +42,7 @@
 // Logger
 static const rclcpp::Logger LOGGER = rclcpp::get_logger("moveit_collision_detection.collision_robot");
 
-static inline bool validateScale(double scale)
+static inline bool validateScale(const double scale)
```
Comment (tylerjw) [inline moveit_core/collision_detection/src/collision_env.cpp:45]: @griswaldbrooks helped me find this: https://abseil.io/tips/109

and this: https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#exception-48

The const here is modifying a definition so it has meaning and is a good change. This looked weird to me but after reading those links I'm on board with this change. I have never met a local variable I didn't want to make const.

**Review by tylerjw — APPROVED** [review]

File: moveit_core/collision_detection/src/collision_matrix.cpp (hunk 1 above)
Comment (bgill92) [reply moveit_core/collision_detection/src/collision_matrix.cpp]: Added

**Comment by tylerjw** [comment]
This one is next for merging, rebased.

**Pull request merged**